### PR TITLE
feat(search): surface explorer views in search results

### PR DESCRIPTION
**Pull request by marcelgerber**
## TODOs

- [x] Implement card-based design
- [x] Do grouping by `explorerSlug`
- [x] Do client-side deduplication based on `[viewTitle, explorerSlug]`
- [x] Add click tracking
- [ ] ~_maybe_ do country name removal (`optionalWords`)~
- [ ] ~_probably_ sort explorer cards with more view items higher up~
- [x] cleanup:
	- [x] remove display of `explorers` index
	- [x] remove `SearchIndexName.explorers` and all uses (#3472)
	- [x] remove all references, styles etc.
	- [x] css styles

### CODE REVIEW

**Comment by marcelgerber** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3429?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3429** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3429?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3428** <a href="https://app.graphite.dev/github/pr/owid/owid-grapher/3428?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @marcelgerber and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: site/search/SearchPanel.tsx [hunk 1]
```diff
@@ -397,21 +571,21 @@ const SearchResults = (props: SearchResultsProps) => {
                     </section>
                 </NoResultsBoundary>
             </Index>
-            <Index indexName={getIndexName(SearchIndexName.Explorers)}>
```
Comment (marcelgerber) [inline site/search/SearchPanel.tsx:400]: The diff here and below is noisy, but it's really just:
- Swapping the indexes for `charts` and `explorers`
- Changing the `cutoffNumber`s
- Using the new `<ExplorerViewHits>` component

File: site/search/SearchPanel.tsx [hunk 2]
```diff
@@ -128,17 +136,152 @@ function ChartHit({ hit }: { hit: IChartHit }) {
     )
 }
 
-function ExplorerHit({ hit }: { hit: IExplorerHit }) {
+interface ExplorerViewHitWithPosition extends IExplorerViewHit {
```
Comment (ikesau) [inline site/search/SearchPanel.tsx:139]: I think this is a regression not from this PR, but it seems like `data-algolia-position` is now only local to each section, i.e. posts start at 1, charts start at 1, etc.

charts were meant to start at `numberOfPosts + 1`

**Review by ikesau — CHANGES_REQUESTED** [review]
The functionality is overall very nice! Just a few comments on behind the scenes stuff

File: site/search/SearchPanel.tsx (hunk 2 above)
Comment (marcelgerber) [reply site/search/SearchPanel.tsx:139]: I think that's what it was meant to do, right?
`data-algolia-position` is the _local_ position, and `globalPosition` gets computed in the click handler on its own (and only gets sent to GTM, not to Algolia).

File: site/search/SearchPanel.tsx (hunk 1 above)
Comment (marcelgerber) [reply site/search/SearchPanel.tsx:399]: The whole "getting 4 sets of results" thing I could finally resolve now (by cheating the Algolia types in a somewhat hacky way 😭).
I then realized that our current Algolia click tracking also didn't really work - the `queryId` is different depending on the index, but we were always sending along the id of the `pages` index.

File: site/search/SearchPanel.tsx [hunk 3]
```diff
@@ -128,17 +136,152 @@ function ChartHit({ hit }: { hit: IChartHit }) {
...
+                            href={`${BAKED_BASE_URL}/${EXPLORERS_ROUTE_FOLDER}/${view.explorerSlug}${view.viewQueryParams}`}
+                            className="search-results__explorer-view-title-container"
+                        >
+                            <Highlight
+                                attribute="viewTitle"
+                                hit={view}
+                                highlightedTagName="strong"
+                                className="search-results__explorer-view-title"
+                            />
+                            <FontAwesomeIcon icon={faArrowRight} />
```
[the remark this answers is not in the document]
Comment (marcelgerber) [reply site/search/SearchPanel.tsx:270]: Could achieve the same thing using `&zwj;` (zero-width joiner) and `white-space: nowrap`.
PR will go up soon, together with other design changes.

**Pull request merged**